pkg/sql/distsqlrun/joinreader.go

```diff
@@ -480,7 +500,9 @@ func (jr *joinReader) collectOutputRows() joinReaderState {
 // prepares for another input batch.
 func (jr *joinReader) emitRow() (joinReaderState, sqlbase.EncDatumRow) {
 	if len(jr.toEmit) == 0 {
-		if jr.finalLookupBatch {
+		// if joinType was LeftSemiJoin, we would have already removed all the matching keys
+		// if no keys are left match skip and read next batch of input
+		if jr.finalLookupBatch || len(jr.keyToInputRowIndices) == 0 {
```

Comment:
@solongordon I added this condition to avoid reading the right side if we know we have already match all left indexes for the batch. Do you  want me to do this or remove it?